src/circuit.rs

```diff
@@ -50,13 +270,401 @@ impl plonk::Circuit<pallas::Base> for Circuit {
+                    nk,
+                    rivk,
+                )?
+            };
+
+            // [ivk] g_d_old
+            g_d_old.mul(layouter.namespace(|| "[ivk] g_d_old"), ivk.inner())?
+        };
+
+        // Old note commitment integrity.
+        let _cm_old = {
+            let old_note_commit_config = config.old_note_commit_config.clone();
+
+            let rcm_old = self.rcm_old.as_ref().map(|rcm_old| **rcm_old);
+
+            // g★_d || pk★_d || i2lebsp_{64}(v) || i2lebsp_{255}(rho) || i2lebsp_{255}(psi)
+            old_note_commit_config.assign_region(
+                layouter.namespace(|| {
+                    "g★_d || pk★_d || i2lebsp_{64}(v) || i2lebsp_{255}(rho) || i2lebsp_{255}(psi)"
+                }),
+                config.sinsemilla_chip_1(),
+                config.ecc_chip(),
+                g_d_old.inner(),
+                pk_d_old.inner(),
+                v_old,
+                rho_old,
+                psi_old,
+                rcm_old,
+            )?
+        };
+
+        // new note commitment integrity.
+        let _cmx = {
+            let new_note_commit_config = config.new_note_commit_config.clone();
+
+            // Witness g_d_new_star
+            let g_d_new = {
+                let g_d_new = self
+                    .g_d_new_star
+                    .map(|bytes| pallas::Affine::from_bytes(&bytes).unwrap());
+                Point::new(
+                    ecc_chip.clone(),
+                    layouter.namespace(|| "witness g_d_new_star"),
+                    g_d_new,
+                )?
+            };
+
+            // Witness pk_d_new_star
+            let pk_d_new = {
+                let pk_d_new = self
+                    .pk_d_new_star
+                    .map(|bytes| pallas::Affine::from_bytes(&bytes).unwrap());
+                Point::new(
+                    ecc_chip,
+                    layouter.namespace(|| "witness pk_d_new"),
+                    pk_d_new,
+                )?
+            };
+
+            // Witness psi_new
+            let psi_new = self.load_private(
+                layouter.namespace(|| "witness psi_new"),
+                config.advices[0],
+                self.psi_new,
+            )?;
+
+            let rcm_new = self.rcm_new.as_ref().map(|rcm_new| **rcm_new);
+
+            // g★_d || pk★_d || i2lebsp_{64}(v) || i2lebsp_{255}(rho) || i2lebsp_{255}(psi)
+            let cm_new = new_note_commit_config.assign_region(
+                layouter.namespace(|| {
+                    "g★_d || pk★_d || i2lebsp_{64}(v) || i2lebsp_{255}(rho) || i2lebsp_{255}(psi)"
+                }),
+                config.sinsemilla_chip_2(),
+                config.ecc_chip(),
+                g_d_new.inner(),
+                pk_d_new.inner(),
+                v_new,
+                *nf_old.inner(),
+                psi_new,
+                rcm_new,
+            )?;
+
+            cm_new.extract_p()
+        };
+
```

Comment:
I don't see the **Enable spend flag** or **Enable output flag** checks.

Reply:
This has been implemented in https://github.com/zcash/orchard/pull/101/commits/e4eda3a67fc5ed598783990934af57c84471a36a.